### ISSUE TITLE
[Bug]: FLASH_ATTN_V100 003出现异常短回复

### ISSUE THREAD

**Issue by yang122001** [1CatAI/1Cat-vLLM-0.0.1#24]
### Your current environment

networks:
  mybridge:
    external: true
services:
  vllm-openai:
    shm_size: '32g'
    image: vllm/vllm-openai:1cat003_fa
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              device_ids: ["0","1","2","3"]
              capabilities: [gpu]
    environment:
      VLLM_SLEEP_WHEN_IDLE: 1
      VLLM_SM70_ENABLE_LM_HEAD_FASTPATH: 1
        # VLLM_LOGGING_LEVEL: DEBUG
        # CUDA_HOME: "/usr/local/cuda-12.8"
        # PATH: "$CUDA_HOME/bin:$PATH"
        # LD_LIBRARY_PATH: "$CUDA_HOME/lib64:${LD_LIBRARY_PATH:-}"
    ports:
      - "8001:8000"
    volumes:
      - ~/models:/home/models
    entrypoint: vllm serve 
    networks:
      - mybridge
    command: >
      --model /home/models/Qwen3.6-27B-AWQ
      --served_model_name Qwen3.5-27B
      --tensor-parallel-size 4
      --dtype float16
      --gpu-memory-utilization 0.80
      --max-model-len 196617
      --max-num-seqs 4
      --max-num-batched-tokens 16384
      --enable-chunked-prefill
      --attention-backend FLASH_ATTN_V100
      --enable-auto-tool-choice 
      --tool-call-parser qwen3_coder
      --default-chat-template-kwargs '{"enable_thinking": false}'
      --compilation-config '{"cudagraph_mode":"full_and_piecewise","cudagraph_capture_sizes":[1,2]}' 
      --skip-mm-profiling
      --enable-prefix-caching
      --enable-log-requests

### 🐛 Describe the bug

执行测速过程中，出现异常短回复：

<img width="1334" height="922" alt="Image" src="https://github.com/user-attachments/assets/75e3ad97-dbe1-4863-8d77-9dfe2e209d04" />

然后在openclaw对话中，发生异常短回复：

<img width="2136" height="918" alt="Image" src="https://github.com/user-attachments/assets/3474567d-b04b-45bb-be20-dcfff603ce66" />

正常应该输出判卷解析

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by bayley** [comment]
Possibly related to #23 ? The bug there also only occurs for some prompts with FLASH_ATTN_V100 enabled.

**Comment by yang122001** [comment]
> Possibly related to [#23](https://github.com/1CatAI/1Cat-vLLM/issues/23) ? The bug there also only occurs for some prompts with FLASH_ATTN_V100 enabled.

Yes, my observation is that it first outputs normally, then abnormally, and finally outputs exclamation marks. It's similar to what you observed.

**Comment by yang122001** [comment]
#23 关联问题

**Comment by yang122001** [comment]
由于部分人打包后没有出现降智情况，我怀疑和包版本有关，现补充我的环境信息，

> nvcc: NVIDIA (R) Cuda compiler driver
> Copyright (c) 2005-2025 NVIDIA Corporation
> Built on Fri_Feb_21_20:23:50_PST_2025
> Cuda compilation tools, release 12.8, V12.8.93
> Build cuda_12.8.r12.8/compiler.35583870_0

> pip: aiohappyeyeballs                  2.6.1
> aiohttp                           3.13.5
> aiosignal                         1.4.0
> annotated-doc                     0.0.4
> annotated-types                   0.7.0
> anthropic                         0.97.0
> anyio                             4.13.0
> apache-tvm-ffi                    0.1.10
> astor                             0.8.1
> attrs                             26.1.0
> blake3                            1.0.8
> cachetools                        7.0.6
> cbor2                             6.0.1
> certifi                           2026.4.22
> cffi                              2.0.0
> charset-normalizer                3.4.7
> click                             8.3.3
> cloudpickle                       3.1.2
> compressed-tensors                0.13.0
> cryptography                      47.0.0
> cuda-bindings                     13.2.0
> cuda-pathfinder                   1.5.4
> cuda-python                       13.2.0
> cupy-cuda12x                      14.0.1
> depyf                             0.20.0
> dill                              0.4.1
> diskcache                         5.6.3
> distro                            1.9.0
> dnspython                         2.8.0
> docstring_parser                  0.18.0
> einops                            0.8.2
> email-validator                   2.3.0
> fastapi                           0.136.1
> fastapi-cli                       0.0.24
> fastapi-cloud-cli                 0.17.1
> fastar                            0.11.0
> filelock                          3.29.0
> flash_attn_v100                   26.2
> flashinfer-python                 0.6.2
> frozenlist                        1.8.0
> fsspec                            2026.4.0
> gguf                              0.18.0
> grpcio                            1.80.0
> grpcio-reflection                 1.80.0
> h11                               0.16.0
> hf-xet                            1.4.3
> httpcore                          1.0.9
> httptools                         0.7.1
> httpx                             0.28.1
> httpx-sse                         0.4.3
> huggingface_hub                   0.36.2
> idna                              3.13
> ijson                             3.5.0
> interegular                       0.3.3
> Jinja2                            3.1.6
> jiter                             0.14.0
> jmespath                          1.1.0
> jsonschema                        4.26.0
> jsonschema-specifications         2025.9.1
> lark                              1.2.2
> llguidance                        1.3.0
> llvmlite                          0.44.0
> lm-format-enforcer                0.11.3
> loguru                            0.7.3
> markdown-it-py                    4.0.0
> MarkupSafe                        3.0.3
> mcp                               1.27.0
> mdurl                             0.1.2
> mistral_common                    1.11.1
> model-hosting-container-standards 0.1.14
> mpmath                            1.3.0
> msgpack                           1.1.2
> msgspec                           0.21.1
> multidict                         6.7.1
> networkx                          3.6.1
> ninja                             1.13.0
> numba                             0.61.2
> numpy                             2.2.6
> nvidia-cublas-cu12                12.8.4.1
> nvidia-cuda-cupti-cu12            12.8.90
> nvidia-cuda-nvrtc-cu12            12.8.93
> nvidia-cuda-runtime-cu12          12.8.90
> nvidia-cudnn-cu12                 9.10.2.21
> nvidia-cudnn-frontend             1.23.0
> nvidia-cufft-cu12                 11.3.3.83
> nvidia-cufile-cu12                1.13.1.3
> nvidia-curand-cu12                10.3.9.90
> nvidia-cusolver-cu12              11.7.3.90
> nvidia-cusparse-cu12              12.5.8.93
> nvidia-cusparselt-cu12            0.7.1
> nvidia-cutlass-dsl                4.4.2
> nvidia-cutlass-dsl-libs-base      4.4.2
> nvidia-ml-py                      13.595.45
> nvidia-nccl-cu12                  2.27.5
> nvidia-nvjitlink-cu12             12.8.93
> nvidia-nvshmem-cu12               3.3.20
> nvidia-nvtx-cu12                  12.8.90
> openai                            2.33.0
> openai-harmony                    0.0.8
> opencv-python-headless            4.13.0.92
> outlines_core                     0.2.11
> packaging                         26.2
> partial-json-parser               0.2.1.1.post7
> pillow                            12.2.0
> pip                               24.0
> prometheus_client                 0.25.0
> prometheus-fastapi-instrumentator 7.1.0
> propcache                         0.4.1
> protobuf                          6.33.6
> psutil                            7.2.2
> py-cpuinfo                        9.0.0
> pybase64                          1.4.3
> pycountry                         26.2.16
> pycparser                         3.0
> pydantic                          2.13.3
> pydantic_core                     2.46.3
> pydantic-extra-types              2.11.1
> pydantic-settings                 2.14.0
> Pygments                          2.20.0
> PyJWT                             2.12.1
> python-dotenv                     1.2.2
> python-json-logger                4.1.0
> python-multipart                  0.0.27
> PyYAML                            6.0.3
> pyzmq                             27.1.0
> ray                               2.55.1
> referencing                       0.37.0
> regex                             2026.4.4
> requests                          2.33.1
> rich                              15.0.0
> rich-toolkit                      0.19.7
> rignore                           0.7.6
> rpds-py                           0.30.0
> safetensors                       0.7.0
> sentencepiece                     0.2.1
> sentry-sdk                        2.58.0
> setproctitle                      1.3.7
> setuptools                        80.10.2
> shellingham                       1.5.4
> six                               1.17.0
> sniffio                           1.3.1
> sse-starlette                     3.4.1
> starlette                         0.52.1
> supervisor                        4.3.0
> sympy                             1.14.0
> tabulate                          0.10.0
> tiktoken                          0.12.0
> tokenizers                        0.22.2
> torch                             2.9.1
> torchaudio                        2.9.1
> torchvision                       0.24.1
> tqdm                              4.67.3
> transformers                      4.57.6
> triton                            3.5.1
> typer                             0.25.1
> typing_extensions                 4.15.0
> typing-inspection                 0.4.2
> urllib3                           2.6.3
> uvicorn                           0.46.0
> uvloop                            0.22.1
> vllm                              0.0.3.dev0+g72bb24e2d.d20260430.cu128
> watchfiles                        1.1.1
> websockets                        16.0
> wheel                             0.42.0
> xgrammar                          0.1.29
> yarl                              1.23.0

**Comment by yangzhuxinyzx** [comment]
这个 issue 是基于 0.0.3 的 `FLASH_ATTN_V100` 路线。当时的异常短回复/感叹号问题和旧 Flash-V100 decode/paged KV 路径有关。

1Cat-vLLM 1.2.0 已经重新接入 Flash-V100 backend，`FLASH_ATTN_V100` 现在覆盖 prefill、decode、paged KV、CUDA graph 和长上下文 decode，并且 README 已更新推荐启动命令。

建议直接升级 1.2.0 wheel 后复测：

https://github.com/1CatAI/1Cat-vLLM/releases/tag/v1.2.0

推荐先用 README 中的 Qwen3.6-27B-AWQ public 命令，不要额外开启旧实验 env。若 1.2.0 仍复现异常短回复，请重新贴 1.2.0 的完整 launch command、请求 body 和返回结果。

**Comment by yangzhuxinyzx** [comment]
Current-source audit maps the historical Qwen3.6-27B-AWQ TP4 symptom to D256/GQA 6:1 Flash-V100 XQA paged decode when the route is eligible. The report lacks current-version token/logit/route artifacts, so there is not enough evidence to safely change XQA, prefill, or CUDA Graph code.

Before a kernel change, reproduce with fixed sampling and compare four routes: Triton baseline, Flash+XQA, Flash scalar decode, and Triton prefill+Flash decode; record first divergent layer, attention finite state, logits top-k, and actual route summary. This issue remains open until that V100 matrix identifies the first divergence.
